### PR TITLE
feat: Add energy chart and filters

### PR DESCRIPTION
**Pull request by MattiasGervilliers**

### CODE REVIEW

File: components/solarPanel/chart/Filters.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import type { Filters } from "~/models/chart/filters";
+import moment from "moment";
+
+const { filters } = defineProps<{
+  filters: Filters;
+}>();
+
+
+let startDateISO = computed({
```
Comment (GuillaumeKusiak) [inline components/solarPanel/chart/Filters.vue:11]: const ?

File: components/solarPanel/chart/Filters.vue [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  get: () => moment(filters.startDate).format("YYYY-MM-DDTHH:mm:ss"),
+  set: (value) => {
+    filters.startDate = new Date(value);
+  },
+});
+
+let endDateISO = computed({
```
Comment (GuillaumeKusiak) [inline components/solarPanel/chart/Filters.vue:18]: const

File: components/solarPanel/EnergyChart.vue [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const filters = ref<Filters>({
+  // today at 00:00
+  //use moment to get the current date minus 7 days
+  startDate: moment().subtract(7, 'days').toDate(),
+  // today at 23:59
+  endDate: new Date(),
+  timeUnit: "HOUR",
+});
+
+let energyDetails = ref<EnergyDetails>({
```
Comment (GuillaumeKusiak) [inline components/solarPanel/EnergyChart.vue:16]: const

File: components/solarPanel/EnergyChart.vue [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<script setup lang="ts">
+
+import type { EnergyDetails } from "~/models/solarPanel/energyDetails";
+import type { Filters } from "~/models/chart/filters";
+import moment from "moment";
+
+const filters = ref<Filters>({
+  // today at 00:00
```
Comment (GuillaumeKusiak) [inline components/solarPanel/EnergyChart.vue:8]: useless commentary ?

File: components/solarPanel/chart/View.vue [hunk 5]
```diff
@@ -0,0 +1,110 @@
+<script setup lang="ts">
+import colors from 'tailwindcss/colors';
+import tailwindConfig from '~/tailwind.config';
+import type { EnergyDetails } from "~/models/solarPanel/energyDetails";
+import { ref, watch } from "vue";
+
+const props = defineProps<{
+  energyDetails: EnergyDetails;
+}>();
+let energyDetails = ref(props.energyDetails);
```
Comment (GuillaumeKusiak) [inline components/solarPanel/chart/View.vue:10]: const

File: components/solarPanel/chart/View.vue [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+const chartKey = ref(0);
+
+function updateChart() {
```
Comment (GuillaumeKusiak) [inline components/solarPanel/chart/View.vue:14]: typing with any is very very BAD !!!!!!!!

File: components/solarPanel/chart/View.vue [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  const differenceData = productionData.map((v: any, i: number) => ({ x: v.x, y: (v.y ?? 0) - (consumptionData[i]?.y ?? 0) }));
+
+  data.datasets[0].data = productionData;
+  data.datasets[1].data = consumptionData;
+  data.datasets[2].data = differenceData;
+
+  chartKey.value++;
+}
+
+const data = {
```
Comment (GuillaumeKusiak) [inline components/solarPanel/chart/View.vue:26]: maybe export the data

**Review by MariusPisto — APPROVED** [review]

**Pull request merged**